llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -231,6 +231,48 @@ static KernelMapEntryScope selectDeviceCodeSplitScopeAutomatically(Module &M) {
   return Scope_PerModule;
 }
 
+// Collect all the dependencies for the function.
+static void collectFunctionCallGraphNodes(llvm::Function *Func,
+                                          SetVector<const GlobalValue *> &GVs) {
+  if (!GVs.count(Func))
+    GVs.insert(Func);
+
+  std::vector<llvm::Function *> Workqueue;
+  Workqueue.push_back(Func);
+
+  while (!Workqueue.empty()) {
+    Function *F = &*Workqueue.back();
```

Comment:
Why such complex construct with `&*` is needed? Shouldn't it be simpler?

```suggestion
    Function *F = Workqueue.back();
```